stage 1 of depreate silent fallback of tuning gemm

henrylhtsang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #147798

Differential Revision: [D70045778](https://our.internmc.facebook.com/intern/diff/D70045778/)

context: 
https://github.com/pytorch/pytorch/issues/147479

For the most part, this should not change the behavior.

For int_mm, I also removed 
```
    # TODO: Re-enable eager mode implementation once cuBLAS is fixed
    if use_cutlass or use_triton_template(layout, enable_int32=True):
        choices = []
```
because I think it is unwanted. 

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

henrylhtsang committed:
stage 1 of depreate silent fallback of tuning gemm



Differential Revision: [D70045778](https://our.internmc.facebook.com/intern/diff/D70045778/)

[ghstack-poisoned]

chenyang78 commented:
It seems to be good to me. Let's get some feedback from @eellison, because it would affect how the user uses a core config. Any though, Elias? Thanks!

henrylhtsang commented:
@eellison ?

henrylhtsang commented:
@pytorchbot merge -i

wdvr commented:
@pytorchmergebot revert -m 'failing internal builds, discussed with author' -c ghfirst

henrylhtsang committed:
Update on "stage 1 of depreate silent fallback of tuning gemm"


Differential Revision: [D70045778](https://our.internmc.facebook.com/intern/diff/D70045778/)

context: 
https://github.com/pytorch/pytorch/issues/147479

For the most part, this should not change the behavior.

For int_mm, I also removed 
```
    # TODO: Re-enable eager mode implementation once cuBLAS is fixed
    if use_cutlass or use_triton_template(layout, enable_int32=True):
        choices = []
```
because I think it is unwanted. 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang commented:
The denylist allowlist test are actually very nice. They bypass the cutlass long compilation problem for unit tests.

henrylhtsang committed:
Update on "stage 1 of depreate silent fallback of tuning gemm"


Differential Revision: [D70045778](https://our.internmc.facebook.com/intern/diff/D70045778/)

context: 
https://github.com/pytorch/pytorch/issues/147479

For the most part, this should not change the behavior.

For int_mm, I also removed 
```
    # TODO: Re-enable eager mode implementation once cuBLAS is fixed
    if use_cutlass or use_triton_template(layout, enable_int32=True):
        choices = []
```
because I think it is unwanted. 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang commented:
@pytorchbot merge -i

henrylhtsang committed:
Update on "stage 1 of depreate silent fallback of tuning gemm"


Differential Revision: [D70045778](https://our.internmc.facebook.com/intern/diff/D70045778/)

context: 
https://github.com/pytorch/pytorch/issues/147479

For the most part, this should not change the behavior.

For int_mm, I also removed 
```
    # TODO: Re-enable eager mode implementation once cuBLAS is fixed
    if use_cutlass or use_triton_template(layout, enable_int32=True):
        choices = []
```
because I think it is unwanted. 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang commented:
@pytorchbot merge -i

henrylhtsang committed:
Update on "stage 1 of depreate silent fallback of tuning gemm"


Differential Revision: [D70045778](https://our.internmc.facebook.com/intern/diff/D70045778/)

context: 
https://github.com/pytorch/pytorch/issues/147479

For the most part, this should not change the behavior.

For int_mm, I also removed 
```
    # TODO: Re-enable eager mode implementation once cuBLAS is fixed
    if use_cutlass or use_triton_template(layout, enable_int32=True):
        choices = []
```
because I think it is unwanted. 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang commented:
bool initialize_block

henrylhtsang commented:
@pytorchbot merge -i

henrylhtsang commented:
@pytorchbot rebase -b viable/strict

henrylhtsang commented:
@pytorchbot merge -i

henrylhtsang commented:
@pytorchbot rebase -b viable/strict

henrylhtsang commented:
@pytorchbot rebase -b main

henrylhtsang committed:
Update

[ghstack-poisoned]

henrylhtsang commented:
@pytorchbot merge
